codes/bits/zetterberg.yml

```diff
@@ -0,0 +1,37 @@
+
+code_id: zetterberg
+physical: bits
+logical: bits
+
+name: 'Zetterberg code'
+introduced: '\cite{doi:10.1109/TIT.1962.1057682}'
+
+
+description: 'The Zetterberg code \(C_s\) is a binary cyclic code of length \(n = 2^{2s}+1\) generated by the
+minimal polynomial \(g_s(x)\) of \(\alpha\) over GF(2), where \(\alpha\) is a primitive \(n^{th}\) root of unity in the field GF\((2^{4s})\).'
+
+protection: 'Correct all weight-2 errors with a small number of exceptions. By making some alterations to the code, one can form codes that correct all weight-2 errors.'
+
+
+features:
+
+
+  rate: 'The rate is given by \(1-\frac{4s}{n}\), which is asymptotically good, but the minimum distance is a constant of either 5 or 6.'
+
+  encoders:
+    - 'Systematic or non-systematic encoding of information word like in all cyclic codes.'
+
+  decoders:
+    - 'Decoding through linear-feedback shift registers.'
+    - 'Kallquist first described an algebraic decoding theorem. A faster version was later provided in \cite{doi:10.1109/18.149509} and further modified in \cite{doi:10.1109/LCOMM.2010.07.100784} which can detect up to weight error of 3.'
+
+realizations:
+  - 'Code used to provide better protection of data transmission with its double error correcting capacity.'
+
+notes:
+  - 'Quasi-perfect, i.e. each n-dimensional binary vector is at most at distance 3 from a codeword.'

```

Comment:
Make code_id: perfect a cousin and explain this there. (I've added defn of quasi-perfect in that entry.)